### PR TITLE
Feature/df 1719 search chfiches occupations

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchService.java [hunk 1]
```diff
@@ -84,12 +89,43 @@ public ElasticsearchBasedChFicheSearchService(ChFicheElasticsearchRepository chF
 	}
 
 	private BoolQueryBuilder createQueryBuilder(ChFicheSearchRequest searchRequest) {
-		if (!hasText(searchRequest.getQuery())) {
+		if (!hasText(searchRequest.getQuery()) && isEmpty(searchRequest.getOccupationCodes())) {
```
Comment (cjpsycos) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchService.java]: I think this is obsolete, due we have new a "must" relation. needs to be checked.

File: infrastructure/src/test/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchServiceTest.java [hunk 2]
```diff
@@ -132,4 +134,62 @@ public void languageBasedSearch() {
...
+        this.chFicheElasticsearchRepository.save(new ChFicheDocument(chFicheWithOccupation(CHF1.id(), TS1, null, occupation("BFS-1"), occupation("BFS-2"))));
+        this.chFicheElasticsearchRepository.save(new ChFicheDocument(chFicheWithOccupation(CHF2.id(), TS2, null, occupation("BFS-3"))));
+        this.chFicheElasticsearchRepository.save(new ChFicheDocument(chFicheWithOccupation(CHF3.id(), TS3, null, occupation("BFS-4", "CHISCO5-1"))));
+        this.chFicheElasticsearchRepository.save(new ChFicheDocument(chFicheWithOccupation(CHF4.id(), TS4, null, occupation("BFS-5"))));
+        this.chFicheElasticsearchRepository.save(new ChFicheDocument(chFicheWithOccupation(CHF5.id(), TS5, null, occupation("BFS-6", "CHISCO5-1"))));
+
+        ChFicheSearchRequest chFicheSearchRequest = new ChFicheSearchRequest()
+                .setQuery("ein") // occurrences in TS1, TS2, TS3
+                .setOccupationCodes(ImmutableSet.of(
+                        new ChFicheSearchRequest.SearchOccupationCode(ChFicheSearchRequest.OccupationCodeType.BFS, "BFS-1"),
```
Comment (cjpsycos) [inline infrastructure/src/test/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchServiceTest.java]: BFS-1..5 an CHISCO5-1..5 could be as constants in the occupationFixture

File: infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchService.java (hunk 1 above)
Comment (emina-str) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchService.java]: removed

File: infrastructure/src/test/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchServiceTest.java (hunk 2 above)
Comment (emina-str) [reply infrastructure/src/test/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/elasticsearch/chfiche/ElasticsearchBasedChFicheSearchServiceTest.java]: added fixture.

**Pull request merged**